### PR TITLE
User timings

### PR DESCRIPTION
**Pull request by chrisnojima**
This adds timeline marks for our sagas/redux stuff/ and engine

This should help us understand whats going on with all our async stuff.

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits 82996ae49a4f...9d62151fe654 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Profile-Your_Profile_Pending_Proof_Menu_DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504905212457-82996ae49a4f-9d62151fe654-linux/Profile-Your_Profile_Pending_Proof_Menu_DNS-82996ae49a4f.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504905212457-82996ae49a4f-9d62151fe654-linux/Profile-Your_Profile_Pending_Proof_Menu_DNS-9d62151fe654.png)  
   Profile-Your_Profile_Pending_Proof_Menu_DNS comparison
</summary>

File: shared/actions/chat/index.js [hunk 1]
```diff
@@ -343,7 +343,7 @@ function* _updateThread({
 }
 
 function subSagaUpdateThread(yourName, yourDeviceName, conversationIDKey) {
-  return function*({thread}) {
+  return function* subSagaUpdateThreadHelper({thread}) {
```
Comment (chrisnojima) [inline shared/actions/chat/index.js]: I've named some anonymous helpers so we can see them better in the tracing

File: shared/desktop/renderer/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 /*
  * The main renderer. Holds the global store. When it changes we send it to the main thread which then sends it out to subscribers
  */
-
+import '../../dev/user-timings'
```
Comment (chrisnojima) [inline shared/desktop/renderer/index.js]: this has to happen asap, it injects and monkeypatches connect

File: shared/app/globals.native.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // @flow
 /* eslint-disable no-native-reassign, no-global-assign, no-extend-native */
 import 'core-js/es6/reflect' // required for babel-plugin-transform-builtin-extend in RN iOS and Android
+import '../dev/user-timings'
```
Comment (chrisnojima) [inline shared/app/globals.native.js]: this has to happen asap, it injects and monkeypatches connect

File: shared/local-debug.native.js [hunk 4]
```diff
@@ -41,24 +41,27 @@ let config: {[key: string]: any} = {
...
   config.forwardLogs = false
+  config.forwardLogs = true
   config.immediateStateLogging = false
+  config.isDevApplePushToken = true
   config.printOutstandingRPCs = true
   config.printRPC = true
   config.printRoutes = true
   config.reactPerf = false
   config.redirectOnLogout = false
-  config.reduxSagaLogger = true
```
Comment (chrisnojima) [inline shared/local-debug.native.js]: lets turn this off, too expensive

**Comment by chrisnojima** [comment]
image

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
yes, it seemed to not affect performance. if it does we can default it off

**Pull request merged**